tornado-unittests/src/main/java/uk/ac/manchester/tornado/unittests/branching/TestLoopConditions.java

```diff
@@ -0,0 +1,544 @@
+/*
+ * Copyright (c) 2024, APT Group, Department of Computer Science,
+ * The University of Manchester.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.manchester.tornado.unittests.branching;
+
+import org.junit.Test;
+import uk.ac.manchester.tornado.api.ImmutableTaskGraph;
+import uk.ac.manchester.tornado.api.TaskGraph;
+import uk.ac.manchester.tornado.api.TornadoExecutionPlan;
+import uk.ac.manchester.tornado.api.annotations.Parallel;
+import uk.ac.manchester.tornado.api.enums.DataTransferMode;
+import uk.ac.manchester.tornado.api.exceptions.TornadoExecutionPlanException;
+import uk.ac.manchester.tornado.api.types.arrays.IntArray;
+import uk.ac.manchester.tornado.api.types.matrix.Matrix2DDouble;
+import uk.ac.manchester.tornado.unittests.common.TornadoTestBase;
+
+import java.util.Random;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * How to test?
+ *
+ * <code>
+ * tornado-test -V uk.ac.manchester.tornado.unittests.branching.TestLoopConditions
+ * </code>
+ */
+public class TestLoopConditions extends TornadoTestBase {
+
+    public static void conditionBeforeSingleLoopReturn(IntArray a, IntArray b, IntArray c) {
+        if (a.get(0) > b.get(0)) {
+            return;
+        }
```

Comment:
What is the expected behaviour of this type of code when running in parallel on a GPU?  We can't stop the GPU kernel.